### DISCUSSION TITLE
"Looking for work"

### DISCUSSION THREAD

**Discussion by joemasilotti** [joemasilotti/railsdevs.com#86] [Ideas]
Some folks added their profile to the site not knowing the full purpose of railsdevs.io. (This one's on me, and hopefully #82 will help.) Because of that, folks might be listed on a site that says "Find Rails developers looking for work" and, well, not be looking for work.

I've drafted #85 to help take care of this. It adds a new column, `developers.looking_for_work`, that can be toggled. If false, then the developer's profile will no longer be displayed on the site.

<img width="1074" alt="image" src="https://user-images.githubusercontent.com/2092156/141411370-aa828ef0-40c8-4f6c-8ac9-a6ff6df90f5a.png">

Ideally, this boolean is non-null. But what should the default be? My first pass is to mark all developer's who are "available" more than a year out as not looking for work. That might not be 100% correct, but I think it should be close enough.

Is there a better heuristic or approach? Technically I could mark everyone as "not looking" and have them manually check the box, but that feels like a lot right now. ...or am I thinking about this the wrong way? Would love to hear what you think!

**Comment by vamsipavanmahesh** [comment]
Looking for work need not be black and white IMHO

Could add some more things like passively looking etc

**Reply by joemasilotti** [reply]
That's a great call. Stack Overflow does a great job with this. We could borrow their options.

<img width="1066" alt="image" src="https://user-images.githubusercontent.com/2092156/141423233-68357a87-cf8d-4b66-a140-74bd91c1ccc1.png">

Anything else you would add there?

**Reply by vamsipavanmahesh** [reply]
This seems like a perfectly well researched one :) Let's go 🚀

**Comment by davegudge** [comment]
Below are a few comments, _related to availability_, that I [originally added](https://github.com/joemasilotti/railsdevs.io/pull/68#discussion_r745938139) to #68

> If the site is also intended for developers who are currently employed full-time, it may be worth having an availability duration option (as well as the current available_on date).

> An availability duration would provider developers, who are currently employed full-time, the ability to represent their notice period without continually having to update the available_on date.

> Perhaps the availability option is determined by if a developers current employment is either full-time vs freelance or contractor. That said, I believe contractors tend to have an end date, thus availability duration may be more suitable.

**Reply by joemasilotti** [reply]
Making sure I completely understand, are you proposing `developers.available_until` to sit alongside `developers.available_on`?

**Reply by davegudge** [reply]
> `developers.available_until`

I _think_ by this you mean, _I'm available now, but only until a certain date (as I have other work already booked in_?  I've not actually considered that scenario.

> _[a field]_ to sit alongside `developers.available_on`?

Yes, another field to sit along side `developers.available_on` is what I was thinking. For example, I am employed full-time so I'm available in 90 days (notice period) if a company is interested in employing me. Perhaps `developers.available_in` which stores a duration to represent a notice period e.g. `developers.available_in = 90.days`.

I do wonder if you could get clever with the UI and only display one of the two availability fields based upon the developers employment type. e.g. If currently employed full-time, `developers.available_in` is probably more suitable, but if you're a freelancer, it's likely you have a set date that you become available thus `developers.available_on` is more suitable? I'm assuming most contracts would have a set end date to?

**Reply by joemasilotti** [reply]
Ahhh, got it! Thanks for clarifying.

Having two columns, but only showing one option in the UI, does sound like it would cover a fair amount of use-cases. Happy to review a PR if you want to submit one. I don't think I'll take this issue on directly as it doesn't seem super high value compared to the other stuff in the backlog.

**Reply by fabionl** [reply]
Would “available for” make sense?

Maybe: part-time contracts, full-time contracts, part-time employment, full-time employment.

For example, an employed developer could still be available to part-time contracts but not to full-time. Or maybe available to full-time contract, but not interested in a full time jobs.

Notes: I’m not native English speaker, and terminology is most likely a bit off.

**Reply by joemasilotti** [reply]
That's a great call @fabionl. It also fits in with the goal of railsdevs better than the Stack Overflow screenshot. Thanks for the input!

**Comment by joemasilotti** [comment]
Combining feedback from folks, here's where I'm at.

<img width="1080" alt="image" src="https://user-images.githubusercontent.com/2092156/141704753-4cd33c06-4e0d-4b7d-bc7a-3eafc6ed3bdd.png">

I feel like this is a huge step forward. Developers can select how aggressively they are looking with an option to say they aren't (right now). And they can indicate to employers what type of role they are looking for. Lastly, they can still set their date when they can first start working.

All of this together should make for a fairly robust profile to show off!

---

I'd love feedback on...

1. The overall approach. Is this too much information for a form?
2. The copy. Do the options, subtitles, and headings make sense?

**Reply by fabionl** [reply]
It looks great!

**Comment by joemasilotti** [comment]
Making progress, folks!

<img width="1089" alt="image" src="https://user-images.githubusercontent.com/2092156/142295446-61ac7404-01f5-44f5-8f6d-6499f8091c8c.png">

**Reply by joemasilotti** [reply]
P.S. PR #85 is updated with this working code.

**Comment by joemasilotti** [comment]
Merged and deployed with #85!
